### PR TITLE
Implement gitlab service jira

### PR DESCRIPTION
**Pull request by jorcau**
Hi,

I'd like to add the Jira's integration implementation (see [doc.](https://docs.gitlab.com/ce/api/services.html#jira)).

Here's a sample of how it can be used:
```hcl
resource "gitlab_project" "awesome_project" {
  name             = "awesome_project"
  description      = "My awesome project."
  visibility_level = "public"
}

resource "gitlab_service_jira" "jira" {
  project  = "${gitlab_project.awesome_project.id}"
  url      = "https://jira.example.com"
  username = "user"
  password = "mypass"
}
```

It includes documentation and acceptance tests.

**Note**: there's currently an issue with the `jira_issue_transition_id` parameter preventing it to work properly (see https://gitlab.com/gitlab-org/gitlab-ce/issues/55109). But the integration still can be used. I'll make the necessary updates to both Gitlab Go client and Terraform provider when the fix is released (probably v11.9)

Thanks,

-------
Edit:
- [ ] Requires PR #143 to be merged

### CODE REVIEW

**Comment by roidelapluie** [comment]
Shouldn't this (and slack) be a block inside gitlab_project?)

**Comment by jorcau** [comment]
> Shouldn't this (and slack) be a block inside gitlab_project?)

I believe this would make the _gitlab\_projet_ resource harder to deal with if it contains all the services both to implement/document and use.

Also, it'd be more complicate to industrialize. Let's say you want to define 2 kinds of Jira service configurations for your projects. If the service is an independent resource, then you can easily create 2 Terraform modules independently of your project configuration. Which, IMO is much more flexible.

**Comment by roidelapluie** [comment]
oh I did not know you culd have 2 per project

**Comment by jorcau** [comment]
> oh I did not know you culd have 2 per project

Not by project but I could have one service configuration for half of my projects and another one for the other half of the projects.

**Comment by rsrchboy** [comment]
...or you could want to create a jira configuration for a `data "gitlab_project" "foo"`, etc.

**Comment by bilby91** [comment]
Anything blocking this one ? Looking forward on provisioning our repos with Jira & TF :)

File: gitlab/resource_gitlab_service_jira.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+				Computed: true,
+			},
+			"active": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)`), "value must be a url"),
```
Comment (roidelapluie) [inline gitlab/resource_gitlab_service_jira.go]: I would think that there is a standard validation for this in terraform?

Comment (roidelapluie) [reply gitlab/resource_gitlab_service_jira.go]: maybe a func with URL.Parse

Comment (jorcau) [reply gitlab/resource_gitlab_service_jira.go]: Done!

**Comment by roidelapluie** [comment]
@jorcau tests are failing

**Comment by jorcau** [comment]
> @jorcau tests are failing

Yep, there is a change in Gitlab 11.9 API that makes go-gitlab fail with Jira. More details on  https://github.com/xanzy/go-gitlab/pull/547#issuecomment-488656569.

I guess we will have to wait for a fix before this MR can be merged.

**Comment by jorcau** [comment]
FYI, I just made a fix on go-gitlab (https://github.com/xanzy/go-gitlab/pull/623) that should enable me to complete this MR soon.

File: gitlab/resource_gitlab_service_jira.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+func resourceGitlabServiceJiraRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	project := d.Get("project").(string)
+
+	log.Printf("[DEBUG] Read Gitlab Jira service %s", d.Id())
+
+	jiraService, response, err := client.Services.GetJiraService(project)
+	if err != nil {
+		if response.StatusCode == 404 {
```
Comment (roidelapluie) [inline gitlab/resource_gitlab_service_jira.go]: please remove this

Comment (roidelapluie) [reply gitlab/resource_gitlab_service_jira.go]: @jorcau mmh can you do this in a new pr?

Comment (jorcau) [reply gitlab/resource_gitlab_service_jira.go]: Done in #147

**Pull request merged**